src/modules/iasql_functions/rpcs/iasql_help.ts

```diff
@@ -0,0 +1,65 @@
+import { IasqlFunctions } from '..';
+import {
+  Context,
+  PostTransactionCheck,
+  PreTransactionCheck,
+  RpcBase,
+  RpcResponseObject,
+} from '../../interfaces';
+
+/**
+ * Method to list IaSQL functions, their description and usage example
+ *
+ * Returns following columns:
+ * - name: function name
+ * - signature: signature used to call the function
+ * - description: a description on what the function does
+ * - sample_usage: sample usage of the function
+ *
+ */
+export class IasqlHelp extends RpcBase {
+  /**
+   * @internal
+   */
+  module: IasqlFunctions;
+  /** @internal */
+  preTransactionCheck = PreTransactionCheck.NO_CHECK;
+  /** @internal */
+  postTransactionCheck = PostTransactionCheck.NO_CHECK;
+  /** @internal */
+  inputTable = {};
+  /**
+   * @internal
+   */
+  outputTable = {
+    name: 'varchar',
+    signature: 'varchar',
+    description: 'varchar',
+    sample_usage: 'varchar',
+  } as const;
+  call = async (
+    dbId: string,
+    _dbUser: string,
+    _ctx: Context,
+  ): Promise<RpcResponseObject<typeof this.outputTable>[]> => {
+    const returnList = [];
+
+    for (const [functionName, documentation] of Object.entries(this.module.provides.functions)) {
```

Comment:
Should this only be for the `iasql_functions` module, or for all modules currently installed by the user?

I think the latter would be more interesting, though we should probably:

1. Have a `module` column in the output to indicate the source module the function comes from.
2. Perhaps put that behavior behind an optional flag that has to be `'all'` or something like that if the output seems too overwhelming?

Reply:
well, we were only showing in `iasql_help()` the functions provided by us in this module. I think that would be a good idea but maybe seems like a follow-up that can be done using this work as base?

Reply:
This is how will it work without an input:
<img width="1481" alt="image" src="https://user-images.githubusercontent.com/5357435/220635530-ed735d61-2a3f-46e9-ac95-476aa9083dc1.png">

Yet, the user can narrow down the result like normal queries:
```sql
SELECT * FROM iasql_help() WHERE module = 'iasql_functions';
```
or
```sql
SELECT * FROM iasql_help() WHERE name LIKE 'iasql_%`;
```

Reply:
Let's leave it as-is for this PR, this is already *much* nicer.